src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,616 @@
+                Snake1d(dimension),
+                WNConv1d(dimension, dimension, kernel_size=1),
+            ]
+        )
+
+    def forward(self, input_tensor):
+        """
+        Forward pass through the residual unit.
+
+        Args:
+            input_tensor (`torch.Tensor` of shape `(batch_size, channels, time_steps)`): 
+                Input tensor .
+
+        Returns:
+            output_tensor (`torch.Tensor` of shape `(batch_size, channels, time_steps)`)
+                Input tensor after passing through the residual unit.
+        """
+        output_tensor = input_tensor
+        for block in self.block: 
+            output_tensor = block(output_tensor)
+
+        padding = (input_tensor.shape[-1] - output_tensor.shape[-1]) // 2
+        if padding > 0:
+            input_tensor = input_tensor[..., padding:-padding]
+        output_tensor = input_tensor + output_tensor
+        return output_tensor
+
+
+class DacEncoderBlock(nn.Module):
+    """Encoder block used in DAC encoder."""
+    def __init__(self, dimension: int = 16, stride: int = 1):
+        super().__init__()
+        self.block = nn.ModuleList(
+            [
+                DacResidualUnit(dimension // 2, dilation=1),
+                DacResidualUnit(dimension // 2, dilation=3),
+                DacResidualUnit(dimension // 2, dilation=9),
+                Snake1d(dimension // 2),
+                WNConv1d(
+                    dimension // 2,
+                    dimension,
+                    kernel_size=2 * stride,
+                    stride=stride,
+                    padding=math.ceil(stride / 2),
+                ),
+            ]
+        )
+
+    def forward(self, input):
+        for block in self.block: 
+            input = block(input)
+        return input
+
+
+class DacDecoderBlock(nn.Module):
+    """Decoder block used in DAC decoder."""
+    def __init__(self, input_dim: int = 16, output_dim: int = 8, stride: int = 1):
+        super().__init__()
+        self.block = nn.ModuleList(
+            [
+                Snake1d(input_dim),
+                WNConvTranspose1d(
+                    input_dim,
+                    output_dim,
+                    kernel_size=2 * stride,
+                    stride=stride,
+                    padding=math.ceil(stride / 2),
+                ),
+                DacResidualUnit(output_dim, dilation=1),
+                DacResidualUnit(output_dim, dilation=3),
+                DacResidualUnit(output_dim, dilation=9),
+            ]
+        )
+
+    def forward(self, input):
+        for block in self.block: 
+            input = block(input)
+        return input
+
+
+class DacResidualVectorQuantize(nn.Module):
+    """
+    ResidualVectorQuantize block - Introduced in SoundStream: An end2end neural audio codec (https://arxiv.org/abs/2107.03312)
+    """
+    def __init__(
+        self,
+        config: DacConfig,
+    ):
+        super().__init__()
+
+        input_dim = config.latent_dim
```

Comment:
There's no point here re-assigning all these config attributes to new variable names -> the config should pass directly to  `DacVectorQuantize`, saving you all this boilerplate